Client/Coordinators/Library/BookmarksCoordinator.swift

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Storage
+
+protocol BookmarksCoordinatorDelegate: AnyObject {
+    func start(from cell: BookmarksFolderCell)
+
+    func showBookmarkDetail(for node: FxBookmarkNode, folder: FxBookmarkNode)
+}
+
+class BookmarksCoordinator: BaseCoordinator, BookmarksCoordinatorDelegate {
+    // MARK: - Properties
+
+    private let profile: Profile
+    private weak var parentCoordinator: LibraryCoordinatorDelegate?
+
+    // MARK: - Initializers
+
+    init(
+        router: Router,
+        profile: Profile,
+        parentCoordinator: LibraryCoordinatorDelegate?
+    ) {
+        self.profile = profile
+        self.parentCoordinator = parentCoordinator
+        super.init(router: router)
+    }
+
+    // MARK: - BookmarksCoordinatorDelegate
+
+    func start(from folderCell: BookmarksFolderCell) {
+        guard let folder = folderCell as? FxBookmarkNode else { return }
```

Comment:
Could we pass the `FxBookmarkNode` directly instead of having a guard?
```suggestion
    func start(from folderCell: FxBookmarkNode) {
```

Reply:
Yeah exactly, I don't think there's another way for this, but would prevent us from guarding in the coordinator at least. I would prefer this in the VC

Reply:
ok perfect